src/vector_index.c

```diff
@@ -164,3 +165,47 @@ const char *VecSimAlgorithm_ToString(VecSimAlgo algo) {
   }
   return NULL;
 }
+
+void VecSim_RdbSave(RedisModuleIO *rdb, VecSimParams *vecsimParams) {
+  RedisModule_SaveUnsigned(rdb, vecsimParams->type);
+  RedisModule_SaveUnsigned(rdb, vecsimParams->size);
+  RedisModule_SaveUnsigned(rdb, vecsimParams->metric);
+  RedisModule_SaveUnsigned(rdb, vecsimParams->algo);
+
+  switch (vecsimParams->algo) {
+  case VecSimAlgo_BF:
+    RedisModule_SaveUnsigned(rdb, vecsimParams->bfParams.initialCapacity);
+    RedisModule_SaveUnsigned(rdb, vecsimParams->bfParams.blockSize);
+    break;
+  case VecSimAlgo_HNSWLIB:
+    RedisModule_SaveUnsigned(rdb, vecsimParams->hnswParams.initialCapacity);
+    RedisModule_SaveUnsigned(rdb, vecsimParams->hnswParams.M);
+    RedisModule_SaveUnsigned(rdb, vecsimParams->hnswParams.efConstruction);
+    RedisModule_SaveUnsigned(rdb, vecsimParams->hnswParams.efRuntime);
+    break;
+  }
+}
+
+int VecSim_RdbLoad(RedisModuleIO *rdb, VecSimParams *vecsimParams, int encver) {
+  vecsimParams->type = LoadUnsigned_IOError(rdb, goto fail);
+  vecsimParams->size = LoadUnsigned_IOError(rdb, goto fail);
+  vecsimParams->metric = LoadUnsigned_IOError(rdb, goto fail);
+  vecsimParams->algo = LoadUnsigned_IOError(rdb, goto fail);
+
+  switch (vecsimParams->algo) {
+  case VecSimAlgo_BF:
+    vecsimParams->bfParams.initialCapacity = LoadUnsigned_IOError(rdb, goto fail);
+    vecsimParams->bfParams.blockSize = LoadUnsigned_IOError(rdb, goto fail);
+    break;
+  case VecSimAlgo_HNSWLIB:
+    vecsimParams->hnswParams.initialCapacity = LoadUnsigned_IOError(rdb, goto fail);
+    vecsimParams->hnswParams.M = LoadUnsigned_IOError(rdb, goto fail);
+    vecsimParams->hnswParams.efConstruction = LoadUnsigned_IOError(rdb, goto fail);
+    vecsimParams->hnswParams.efRuntime = LoadUnsigned_IOError(rdb, goto fail);
+    break;
+  }
```

Comment:
default return RedisModule_ERR?

Reply:
The enum has only 2 variants.
Using default disables warning at compilation